elyra/pipeline/kfp/kfp_authentication.py

```diff
@@ -0,0 +1,693 @@
+                         f'response URL: {request_entry[1].url}'
+            else:
+                output = f'{output}\n'\
+                         f'Request URL: {request_entry[0]} '\
+                         f'HTTP status code: {request_entry[1].status_code} '\
+                         f'response URL: {request_entry[1].url}'
+        return output
+
+
+class KFPAuthenticator():
+    """
+    Use this class to authenticate with Kubeflow Pipelines. The authenticate
+    method delegates the actual authentication to an implementation of the
+    AbstractAuthenticator class.
+    """
+
+    def authenticate(self,
+                     api_endpoint: str,
+                     auth_type: str,
+                     runtime_config_name: str,
+                     auth_parm_1: Optional[str] = None,
+                     auth_parm_2: Optional[str] = None) -> Dict[str, Any]:
+        """
+        Try to authenticate with Kubeflow using the provided information.
+
+        :param api_endpoint: Kubeflow Pipelines endpoint URL, as specified in the runtime configuration
+        :type api_endpoint: str
+        :param auth_type Identifies the authentication type to be performed. If the provided value
+         is in the SupportedAuthProviders enum, authentication is performed.
+        :type auth_type: str
+        :param runtime_config_name: Runtime configuration name where kf_endpoint is specified.
+        :type runtime_config_name: str
+        :param auth_parm_1: First authorization parameter from the runtime config, defaults to None
+        :type auth_parm_1: Optional[str], optional
+        :param auth_parm_2: Second authorization parameter from the runtime config, defaults to None
+        :type auth_parm_2: Optional[str], optional
+        :raises AuthenticationError: Authentication failed due to the provided reason.
+        :return: A data structure containing information that enables kfp.Client to connect to api_endpoint
+        :rtype: Dict[str, str]
+        """
+
+        kf_url = urlsplit(api_endpoint)._replace(path='').geturl()
+
+        # return data structure for successful requests
+        auth_info = {
+            'api_endpoint': kf_url,   # KF API endpoint, source: runtime config
+            'auth_type': None,        # Authentication type, source: runtime config
+            'kf_secured': False,      # Indicates whether KF API is secured
+            'cookies': None,          # passed to KFP SDK client as "cookies" param value
+            'existing_token': None    # passed to KFP SDK client as "existing_token" param value
+        }
+
+        try:
+            auth_info['auth_type'] = auth_type
+
+            # Process the authentication request using the appropriate authenticator
+            # implementation. Refer to the class definitions for information how
+            # the request is processed
+
+            if auth_type == SupportedAuthProviders.NO_AUTHENTICATION.value:
```

Comment:
Then compare the types.
```suggestion
            if auth_type == SupportedAuthProviders.NO_AUTHENTICATION:
```